src/client.ts

```diff
@@ -170,15 +180,33 @@ export class Client {
         return new Client(this.session, this.resolver, this.rootDocument, null, null, this.configuration);
     }
 
-    async switchToSpace(spaceId: string): Promise<void> {
+    async switchToSpace(spaceIdOrName: string): Promise<void> {
         if (this.rootDocument === null) {
             throw new Error(
                 "Root document is null; this document is required for the API client. Please ensure that the API endpoint is accessible along with its root document."
             );
         }
 
-        this.spaceId = spaceId;
-        this.spaceRootDocument = await this.get<SpaceRootResource>(this.rootDocument.Links["SpaceHome"], { spaceId: this.spaceId });
+        const spaceList = await this.get<PagingCollection<SpaceResource>>(this.rootDocument.Links["Spaces"]);
+        const uppercaseSpaceIdOrName = toUpper(spaceIdOrName);
```

Comment:
We should be able to get away with just using string.toUpperCase here, not necessary to import the `toUpper` helper function.

```suggestion
        const uppercaseSpaceIdOrName = spaceIdOrName.toUpperCase();
```